diff -ru debian/control debian.new/control
--- debian/control	2019-02-10 17:13:53.000000000 +0100
+++ debian.new/control	2021-07-22 22:12:34.605524214 +0200
@@ -30,7 +30,7 @@
 Multi-Arch: foreign
 Depends: ${shlibs:Depends}, ${misc:Depends}
 Recommends: manpages
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: Binaries
  This package contains utility programs related to the GNU C Library.
  .
@@ -49,7 +49,7 @@
 Multi-Arch: foreign
 Depends: ${shlibs:Depends}, ${misc:Depends}
 Recommends: manpages, manpages-dev
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: Development binaries
  This package contains utility programs related to the GNU C Library
  development package.
@@ -62,7 +62,7 @@
 Depends: ${misc:Depends}
 Breaks: locales (<< 2.21-1)
 Replaces: locales (<< 2.21-1)
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: localization files
  This package contains the translation files for the GNU C library and
  utility programs.
@@ -74,7 +74,7 @@
 Multi-Arch: foreign
 Depends: ${misc:Depends}
 Suggests: glibc-doc-reference
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: Documentation
  Contains man pages for libpthread functions and the complete GNU C Library
  ChangeLog.  The GNU C Library Reference manual has been moved into
@@ -90,7 +90,7 @@
 Breaks: cross-toolchain-base (<< 29~), cross-toolchain-base-ports (<< 25~)
 Conflicts: eglibc-source
 Replaces: eglibc-source
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: sources
  This package contains the sources and patches which are needed to
  build glibc.
@@ -101,7 +101,7 @@
 Priority: standard
 Depends: libc-bin (>> 2.28), libc-l10n (>> 2.28), ${misc:Depends}, debconf | debconf-2.0
 Replaces: manpages-fr-extra (<< 20141022)
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: National Language (locale) data [support]
  Machine-readable data files, shared objects and programs used by the
  C library for localization (l10n) and internationalization (i18n) support.
@@ -119,7 +119,7 @@
 Multi-Arch: foreign
 Depends: libc-l10n (>> 2.28), ${misc:Depends}
 Breaks: locales (<< 2.13-17)
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: Precompiled locale data
  This package contains the precompiled locale data for all supported locales.
  A better alternative is to install the locales package and only select
@@ -132,7 +132,7 @@
 Multi-Arch: foreign
 Priority: optional
 Depends: lsb-base (>= 3.0-6), ${shlibs:Depends}, ${misc:Depends}
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: Name Service Cache Daemon
  A daemon which handles passwd, group and host lookups
  for running programs and caches the results for the next
@@ -152,14 +152,14 @@
          libc0.3 (>= 2.13-5) [hurd-i386],
          libc0.1 (>= 2.13-5) [kfreebsd-i386],
 Priority: optional
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: Transitional package to ensure multiarch compatibility
  This is a transitional package used to ensure multiarch support is present
  in ld.so before unpacking libraries to the multiarch directories.  It can
  be removed once nothing on the system depends on it.
 
 Package: libc6
-Architecture: amd64 arm64 armel armhf hppa i386 m68k mips mipsel mipsn32 mipsn32el mips64 mips64el mipsr6 mipsr6el mipsn32r6 mipsn32r6el mips64r6 mips64r6el nios2 powerpc powerpcspe ppc64 ppc64el riscv64 sparc sparc64 s390x sh3 sh4 x32
+Architecture: amd64
 Section: libs
 Priority: optional
 Multi-Arch: same
@@ -191,7 +191,7 @@
  and the standard math library, as well as many others.
 
 Package: libc6-dev
-Architecture: amd64 arm64 armel armhf hppa i386 m68k mips mipsel mipsn32 mipsn32el mips64 mips64el mipsr6 mipsr6el mipsn32r6 mipsn32r6el mips64r6 mips64r6el nios2 powerpc powerpcspe ppc64 ppc64el riscv64 sparc sparc64 s390x sh3 sh4 x32
+Architecture: amd64
 Section: libdevel
 Priority: optional
 Multi-Arch: same
@@ -206,26 +206,26 @@
  and link programs which use the standard C library.
 
 Package: libc6-dbg
-Architecture: amd64 arm64 armel armhf hppa i386 m68k mips mipsel mipsn32 mipsn32el mips64 mips64el mipsr6 mipsr6el mipsn32r6 mipsn32r6el mips64r6 mips64r6el nios2 powerpc powerpcspe ppc64 ppc64el riscv64 sparc sparc64 s390x sh3 sh4 x32
+Architecture: amd64
 Section: debug
 Priority: optional
 Multi-Arch: same
 Provides: libc-dbg
 Depends: libc6 (= ${binary:Version}), ${misc:Depends}
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: detached debugging symbols
  This package contains the detached debugging symbols for the GNU C
  library.
 
 Package: libc6-pic
-Architecture: amd64 arm64 armel armhf hppa i386 m68k mips mipsel mipsn32 mipsn32el mips64 mips64el mipsr6 mipsr6el mipsn32r6 mipsn32r6el mips64r6 mips64r6el nios2 powerpc powerpcspe ppc64 ppc64el riscv64 sparc sparc64 s390x sh3 sh4 x32
+Architecture: amd64
 Section: libdevel
 Priority: optional
 Multi-Arch: same
 Conflicts: libc-pic
 Provides: libc-pic, glibc-pic
 Depends: libc6 (= ${binary:Version}), ${misc:Depends}
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: PIC archive library
  Contains an archive library (ar file) composed of individual shared objects.
  This is used for creating a library which is a smaller subset of the
@@ -234,11 +234,11 @@
 
 Package: libc6-udeb
 Package-Type: udeb
-Architecture: amd64 arm64 armel armhf hppa i386 m68k mips mipsel mipsn32 mipsn32el mips64 mips64el mipsr6 mipsr6el mipsn32r6 mipsn32r6el mips64r6 mips64r6el nios2 powerpc powerpcspe ppc64 ppc64el riscv64 sparc sparc64 s390x sh3 sh4 x32
+Architecture: amd64
 Section: debian-installer
 Priority: optional
 Provides: libc6, libc-udeb, libnss-dns-udeb, libnss-files-udeb, ${locale-compat:Depends}
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: Shared libraries - udeb
  Contains the standard libraries that are used by nearly all programs on
  the system. This package includes shared versions of the standard C library
@@ -248,7 +248,7 @@
  installer.  Do not install it on a normal system.
 
 Package: libc6.1
-Architecture: alpha ia64
+Architecture: 
 Section: libs
 Priority: optional
 Multi-Arch: same
@@ -273,14 +273,14 @@
           libc6-mipsn32 [mipsn32 mipsn32el],
           libc6-mips32 [mips mipsel],
           hurd (<< 20120408-3) [hurd-i386]
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: Shared libraries
  Contains the standard libraries that are used by nearly all programs on
  the system. This package includes shared versions of the standard C library
  and the standard math library, as well as many others.
 
 Package: libc6.1-dev
-Architecture: alpha ia64
+Architecture: 
 Section: libdevel
 Priority: optional
 Multi-Arch: same
@@ -295,26 +295,26 @@
  and link programs which use the standard C library.
 
 Package: libc6.1-dbg
-Architecture: alpha ia64
+Architecture: 
 Section: debug
 Priority: optional
 Multi-Arch: same
 Provides: libc-dbg
 Depends: libc6.1 (= ${binary:Version}), ${misc:Depends}
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: detached debugging symbols
  This package contains the detached debugging symbols for the GNU C
  library.
 
 Package: libc6.1-pic
-Architecture: alpha ia64
+Architecture: 
 Section: libdevel
 Priority: optional
 Multi-Arch: same
 Conflicts: libc-pic
 Provides: libc-pic, glibc-pic
 Depends: libc6.1 (= ${binary:Version}), ${misc:Depends}
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: PIC archive library
  Contains an archive library (ar file) composed of individual shared objects.
  This is used for creating a library which is a smaller subset of the
@@ -323,11 +323,11 @@
 
 Package: libc6.1-udeb
 Package-Type: udeb
-Architecture: alpha ia64
+Architecture: 
 Section: debian-installer
 Priority: optional
 Provides: libc6.1, libc-udeb, libnss-dns-udeb, libnss-files-udeb, ${locale-compat:Depends}
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: Shared libraries - udeb
  Contains the standard libraries that are used by nearly all programs on
  the system. This package includes shared versions of the standard C library
@@ -362,7 +362,7 @@
           libc6-mipsn32 [mipsn32 mipsn32el],
           libc6-mips32 [mips mipsel],
           hurd (<< 20120408-3) [hurd-i386]
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: Shared libraries
  Contains the standard libraries that are used by nearly all programs on
  the system. This package includes shared versions of the standard C library
@@ -390,7 +390,7 @@
 Multi-Arch: same
 Provides: libc-dbg
 Depends: libc0.3 (= ${binary:Version}), ${misc:Depends}
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: detached debugging symbols
  This package contains the detached debugging symbols for the GNU C
  library.
@@ -403,7 +403,7 @@
 Conflicts: libc-pic
 Provides: libc-pic, glibc-pic
 Depends: libc0.3 (= ${binary:Version}), ${misc:Depends}
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: PIC archive library
  Contains an archive library (ar file) composed of individual shared objects.
  This is used for creating a library which is a smaller subset of the
@@ -416,7 +416,7 @@
 Section: debian-installer
 Priority: optional
 Provides: libc0.3, libc-udeb, libnss-dns-udeb, libnss-files-udeb, ${locale-compat:Depends}
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: Shared libraries - udeb
  Contains the standard libraries that are used by nearly all programs on
  the system. This package includes shared versions of the standard C library
@@ -451,7 +451,7 @@
           libc6-mipsn32 [mipsn32 mipsn32el],
           libc6-mips32 [mips mipsel],
           hurd (<< 20120408-3) [hurd-i386]
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: Shared libraries
  Contains the standard libraries that are used by nearly all programs on
  the system. This package includes shared versions of the standard C library
@@ -479,7 +479,7 @@
 Multi-Arch: same
 Provides: libc-dbg
 Depends: libc0.1 (= ${binary:Version}), ${misc:Depends}
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: detached debugging symbols
  This package contains the detached debugging symbols for the GNU C
  library.
@@ -492,7 +492,7 @@
 Conflicts: libc-pic
 Provides: libc-pic, glibc-pic
 Depends: libc0.1 (= ${binary:Version}), ${misc:Depends}
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: PIC archive library
  Contains an archive library (ar file) composed of individual shared objects.
  This is used for creating a library which is a smaller subset of the
@@ -505,7 +505,7 @@
 Section: debian-installer
 Priority: optional
 Provides: libc0.1, libc-udeb, libnss-dns-udeb, libnss-files-udeb, ${locale-compat:Depends}
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: Shared libraries - udeb
  Contains the standard libraries that are used by nearly all programs on
  the system. This package includes shared versions of the standard C library
@@ -824,7 +824,7 @@
 Multi-Arch: same
 Pre-Depends: libc6 (= ${binary:Version})
 Depends: ${misc:Depends}
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: Shared libraries [Xen version]
  Contains the standard libraries that are used by nearly all programs on
  the system. This package includes shared versions of the standard C
@@ -840,7 +840,7 @@
 Multi-Arch: same
 Pre-Depends: libc0.3 (= ${binary:Version})
 Depends: ${misc:Depends}
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: Shared libraries [Xen version]
  Contains the standard libraries that are used by nearly all programs on
  the system. This package includes shared versions of the standard C
@@ -856,7 +856,7 @@
 Multi-Arch: same
 Pre-Depends: libc6.1 (= ${binary:Version})
 Depends: ${misc:Depends}
-Build-Profiles: <!stage1>
+Build-Profiles: <!stage1 !ioinstr>
 Description: GNU C Library: Shared libraries (EV67 optimized)
  Contains the standard libraries that are used by nearly all programs on
  the system. This package includes shared versions of the standard C
Seulement dans debian.new/control.in: libc0.1
Seulement dans debian.new/control.in: libc0.3
Seulement dans debian.new/control.in: libc6
Seulement dans debian.new/control.in: libc6.1
Seulement dans debian.new/patches/all: ioinstr.diff
diff -ru debian/patches/series debian.new/patches/series
--- debian/patches/series	2019-02-28 00:09:00.000000000 +0100
+++ debian.new/patches/series	2021-07-22 21:22:38.883843149 +0200
@@ -160,3 +160,5 @@
 any/submitted-resolv-unaligned.diff
 any/local-cudacc-float128.diff
 any/git-libio-stdout-putc.diff
+
+all/ioinstr.diff
diff -ru debian/rules debian.new/rules
--- debian/rules	2019-02-10 17:13:53.000000000 +0100
+++ debian.new/rules	2021-07-22 21:21:44.832353070 +0200
@@ -98,7 +98,7 @@
 BASE_MIG = mig
 DEB_GCC_VERSION ?= -8
 
-RUN_TESTSUITE = yes
+RUN_TESTSUITE = no
 TIMEOUTFACTOR = 25
 
 # Set cross and native compiler names, including version.
@@ -136,13 +136,14 @@
     DEB_BUILD_OPTIONS+=nocheck
   endif
 else
-  DEB_ARCH_REGULAR_PACKAGES = $(libc) $(libc)-dev $(libc)-dbg $(libc)-pic libc-bin libc-dev-bin multiarch-support
-  DEB_INDEP_REGULAR_PACKAGES = glibc-doc glibc-source libc-l10n locales
-  DEB_UDEB_PACKAGES = $(libc)-udeb
+  DEB_ARCH_REGULAR_PACKAGES = $(libc)
+  ##$(libc)-dev $(libc)-dbg $(libc)-pic libc-bin libc-dev-bin multiarch-support
+  ##DEB_INDEP_REGULAR_PACKAGES = glibc-doc glibc-source libc-l10n locales
+  ##DEB_UDEB_PACKAGES = $(libc)-udeb
   ## Locales can only be pre-generated during native compiles
-  ifeq ($(DEB_HOST_ARCH),$(DEB_BUILD_ARCH))
-    DEB_ARCH_REGULAR_PACKAGES += locales-all
-  endif
+  ##ifeq ($(DEB_HOST_ARCH),$(DEB_BUILD_ARCH))
+  ##  DEB_ARCH_REGULAR_PACKAGES += locales-all
+  ##endif
 endif
 
 # Generic kernel version check
diff -ru debian/rules.d/control.mk debian.new/rules.d/control.mk
--- debian/rules.d/control.mk	2019-02-10 17:13:53.000000000 +0100
+++ debian.new/rules.d/control.mk	2021-07-22 21:20:07.181224860 +0200
@@ -1,10 +1,11 @@
 libc_packages := libc6 libc6.1 libc0.1 libc0.3
 libc0_1_archs := kfreebsd-amd64 kfreebsd-i386
 libc0_3_archs := hurd-i386
-libc6_archs   := amd64 arm64 armel armhf hppa i386 m68k mips mipsel mipsn32 mipsn32el mips64 mips64el mipsr6 mipsr6el \
-                 mipsn32r6 mipsn32r6el mips64r6 mips64r6el nios2 powerpc powerpcspe ppc64 ppc64el riscv64 \
-                 sparc sparc64 s390x sh3 sh4 x32
-libc6_1_archs := alpha ia64
+libc6_archs   := amd64
+#arm64 armel armhf hppa i386 m68k mips mipsel mipsn32 mipsn32el mips64 mips64el mipsr6 mipsr6el \
+#                mipsn32r6 mipsn32r6el mips64r6 mips64r6el nios2 powerpc powerpcspe ppc64 ppc64el riscv64 \
+#                 sparc sparc64 s390x sh3 sh4 x32
+#libc6_1_archs := alpha ia64
 
 control_deps := $(wildcard debian/control.in/*) $(addprefix debian/control.in/, $(libc_packages))
 triggers := binutils, linux-libc-dev [linux-any], $(BASE_CC)$(DEB_GCC_VERSION)
